src/streamlink_cli/argparser.py

```diff
@@ -261,10 +269,14 @@ def keyvalue(value):
     """
 )
 general.add_argument(
-    "--no-version-check",
-    action="store_true",
+    "--auto-version-check",
+    type=boolean,
+    metavar="{yes,true,1,on,no,false,0,off}",
+    default=False,
```

Comment:
By default should this be true? (having streamlink check the version on each usage by default)